content/docs/user-guide/testing/getting-started/_index.md

```diff
@@ -0,0 +1,277 @@
+
+C++ production code for an O3DE feature gets built into a library, which is later loaded into an application. When writing C++ tests for this code, the test-code should similarly be built into its own separate library. This test-library will then declare a dependency on loading the production library it tests. Building separate test-only libraries has the advantage of keeping production binaries lean and easy to ship, as well as free of extraneous ```#if defined``` blocks for test-only logic. This also ensures tests target the exact same interfaces which ship with the product.
+
+### Registering new C++ tests
+
+While reading the upcoming steps, refer to the following `CMakeLists.txt` example:
+
+```
+# Preexisting module registration from completing the Prerequisite step
+ly_add_target(
+    NAME MyModuleName
+    NAMESPACE Gem
+    FILES_CMAKE
+        mymodule_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            Source
+        PUBLIC
+            Include
+    BUILD_DEPENDENCIES
+        PRIVATE
+            Legacy::CryCommon
+    COMPONENT
+        MyModuleComponent
+)
+
+# New test module registration from Step 1
+# this configures CMake to also build a library called MyModuleName.Tests.so (.dll on Windows)
+# the rest of CMake will refer to this freshly created module by its NAME ("MyModuleName.Tests")
+if(PAL_TRAIT_BUILD_SUPPORTS_TESTS)  # only create these test modules if the target platform supports testing
+
+    ly_add_target(
+        NAME MyModuleName.Tests SHARED
+        NAMESPACE Gem
+        FILES_CMAKE
+            mymodule_test_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Tests
+                Source
+        BUILD_DEPENDENCIES
+            PRIVATE
+                AZ::AzTest
+                Legacy::CryCommon
+                Gem::MyModuleName
+        COMPONENT
+            MyModuleComponent
+    )
+
+    # Register new test module from Step 2
+    ly_add_googletest(
+        NAME MyModuleName.Tests
+    )
+
+endif()
+```
+
+#### Prerequisite: Add a production build target
+
+Before configuring tests, you must first define the library that you want to test. If your tests target an existing feature, then this step is already completed and you simply need to find the correct file path. Target libraries are defined in a `CMakeLists.txt`, which typically lives in the same directory as the code or in a parent directory. For more information on configuring `CMakeLists.txt` files, please refer to [Getting Started with the CMake Build System](/docs/user-guide/build/). Note that you only need to define the production library that the tests target, not complete every feature in the production library. You can start configuring and writing tests before you finish working on the production code.
```

Comment:
The title of the index page for that section was used for the link title. I'm fine accepting this change, as it will be more resilient to changes to the linked page.